pandas/core/indexes/datetimelike.py

```diff
@@ -854,6 +854,116 @@ def _concat_same_dtype(self, to_concat, name):
+        return left._slice_by_value(start, end)
+
+    def _intersect_descending(self, other, **empty_params):
+        # this is essentially a flip of _intersect_ascending
+        if self[0] >= other[0]:
+            left, right = self, other
+        else:
+            left, right = other, self
+
+        start = min(left[0], right[0])
+        end = right[-1]
+
+        if end > start:
+            return type(self)(data=[], **empty_params)
+        return left._slice_by_value(start, end)
+
+    def _offsets_equal(self, other):
+        offsets_equal = True
+        self_offset = getattr(self, 'offset', None)
+        other_offset = getattr(other, 'offset', None)
+        if self_offset is None or other_offset is None or \
+                self_offset != other_offset or other_offset.isAnchored():
+            offsets_equal = False
+        return offsets_equal
+
+    def intersection(self, other):
+        """
+        Specialized intersection for DateTimeIndexOpsMixin objects.
+        May be much faster than Index.intersection.
+
+        Parameters
+        ----------
+        other : Index or array-like
+
+        Returns
+        -------
+        Index
+            A shallow copied intersection between the two things passed in
+        """
+        self._assert_can_do_setop(other)
+
+        if self.equals(other):
+            return self._get_consensus_name(other)
+
+        if not isinstance(other, DatetimeIndexOpsMixin):
+            try:
+                other = DatetimeIndexOpsMixin(other)
+            except (TypeError, ValueError):
+                pass
+            result = Index.intersection(self, other)
+            return result
+        elif (self._offsets_equal(other) or
+              (not self.is_strictly_monotonic or
+                not other.is_strictly_monotonic)):
+            result = Index.intersection(self, other)
+            tz = getattr(self, 'tz', None)
+            result = self._shallow_copy(result._values, name=result.name,
+                                        tz=tz, freq=None)
+            if result.freq is None:
+                result.offset = frequencies.to_offset(result.inferred_freq)
+            return result
+
+        lenghts = len(self), len(other)
+        if lenghts[0] == 0:
+            return self
+        if lenghts[1] == 0:
+            return other
+
+        # handle intersecting things like this
+        # idx1 = pd.to_timedelta((1, 2, 3, 4, 5, 6, 7, 8), unit='s')
+        # idx2 = pd.to_timedelta((2, 3, 4, 8), unit='s')
+        if lenghts[0] != lenghts[1] and (
+                    max(self) != max(other) or min(self) != min(other)):
+            return Index.intersection(self, other)
+
+        # coerce into same order
+        self_ascending = self.is_monotonic_increasing
+        if self_ascending != other.is_monotonic_increasing:
+            other = other.sort_values(ascending=self_ascending)
+
+        # Thanks, PeriodIndex
```

Comment:
huh?

Reply:
This can go away now since with the changes to intersect_ascending and intersect_descending not needing to create an empty index of type(self).